### ISSUE TITLE
draw_text and text_box have different coordinates origin

### ISSUE THREAD

**Issue by paniko0** [prawnpdf/prawn#1123]
Hey guys, I am having a weird behaviour after moving from `draw_text` to `text_box`. We needed to make that change because `draw_text` doesn't support fallback fonts.

If you run the following code, you will notice that the rendered PDF plots the text in different positions:

```ruby
Prawn::Document.generate("a.pdf") do |document|
  document.stroke_axis
  document.text_box 'text', at: [200, 400]
  document.draw_text 'draw', at: [200, 400]
end
```

![image](https://user-images.githubusercontent.com/928351/62985999-991a1400-bdee-11e9-8653-f112f572599d.png)

Apparently, they use different origins, which cause them to render the text in different positions. If I  adjust the `y` of the text_box by 70%, then it would render in the right position:

```ruby
Prawn::Document.generate("a.pdf") do |document|
  document.stroke_axis
  document.text_box 'text', at: [200, 400 + (0.7 * document.font_size)]
  document.draw_text 'draw', at: [200, 400]
end
```

![image](https://user-images.githubusercontent.com/928351/62986024-adf6a780-bdee-11e9-851d-5ff2275f269e.png)

Any ideas on how to elegantly fix this? Or just adjusting it by 70% works?

**Comment by gettalong** [comment]
Haven't looked at the code but I guess the difference is because of different text origins. 

With `#draw_text` you define the lower-left corner where drawing should be done since this method is basically a one-to-one mapping of the PDF text drawing command. Therefore the word "draw" appears above the 400 line.

With `#text_box` you use a high-level method that defines a box into which text should be drawn. And the `at` option defines the upper-left corner of this box (see the [API spec](http://prawnpdf.org/api-docs/2.0/Prawn/Text.html#text_box-instance_method), therefore the text appears below the 400 line.

**Comment by paniko0** [comment]
Thanks for getting back @gettalong 

Any specific reason on why those two methods use two different origins? It seems confusing to me that one uses top-down and the other bottom-up. Do you think it's worth to add an option that specifies the origin so we can have both methods behaving the same way?

Or do you think it's possible to add support to fallback fonts in the draw-text method? I'm opened to implement that if you think that's feasible.

Cheers,

**Comment by gettalong** [comment]
The reason is that these two methods do different things. As I mentioned `#draw_text` is a low-level drawing method, doing nothing more than outputting the PDF code needed. The `#text_box` method, however, is a high-level drawing method, providing, among other things, automatic line wrapping and limiting the drawn text to a rectangular area.

Therefore it doesn't really make sense for the `:at` option to behave exactly the same for these two methods.

As for fallback fonts in `#draw_text`: This may be a solution for your problem.

**Comment by petergoldstein** [comment]
@paniko0 and @gettalong , based on these comments it seems like this issue should be closed.  It may be desirable to open a followup issue to add fallback fonts to `#draw_text`.  Is that a correct summary of the current situation?

**Comment by gettalong** [comment]
@petergoldstein Yes, I think it can be closed. But no, I don't think providing fallback font support in `#draw_text` should be done in a general way, i.e. in Prawn itself, since this is a low-level method with a narrow focus on doing just its thing.

**Comment by petergoldstein** [comment]
@gettalong Ok, thanks.

@pointlessone It sounds like this can just be closed with no follow up.
